app/javascript/mastodon/reducers/contexts.ts

```diff
@@ -0,0 +1,156 @@
+import type { Reducer } from '@reduxjs/toolkit';
+import type { List as ImmutableList } from 'immutable';
+
+import { timelineDelete } from 'mastodon/actions/timelines_typed';
+import type { ApiRelationshipJSON } from 'mastodon/api_types/relationships';
+import type {
+  ApiStatusJSON,
+  ApiContextJSON,
+} from 'mastodon/api_types/statuses';
+import type { Status } from 'mastodon/models/status';
+
+import { blockAccountSuccess, muteAccountSuccess } from '../actions/accounts';
+import { fetchContext } from '../actions/statuses';
+import { TIMELINE_UPDATE } from '../actions/timelines';
+import { compareId } from '../compare_id';
+
+interface State {
+  inReplyTos: Record<string, string | undefined>;
```

Comment:
Can it actually be `undefined`? Aren't we storing only those that are definitely in reply to something?

Reply:
But what if you access the map with an id for which there is no entry? I don't think the `Record` type automatically accounts for that...

Reply:
Hm, something sounds wrong. If you try to access the map with an `id` for which there is no entry, there is no entry. If that's not allowed, `string | undefined` will only help in that you need to have a record that exists but with an `undefined` value, which we can't ensure (and which we don't attempt to ensure, as we skip the entry entirely in https://github.com/mastodon/mastodon/pull/34506/files#diff-7ffc5437fa6af880551a43e51a86a24c5e5bacba0220f315d3441bd5760a3038R42 instead of setting it to `undefined`